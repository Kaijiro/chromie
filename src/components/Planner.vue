<template>
  Here we go, let's go on planning.<br/>
  Planning for {{encounter.name}}<br/>
  {{encounter}}
  <table>
    <tr>
      <td>
        <ul>
          <li v-for="player in encounter.fighters">{{player}}</li>
        </ul>
      </td>
    </tr>
  </table>
</template>

<script lang="ts">
import {defineComponent} from "vue";
import {mapGetters} from "vuex";

export default defineComponent({
  computed: mapGetters({
    encounter: 'selectedEncounter'
  })
});
</script>
