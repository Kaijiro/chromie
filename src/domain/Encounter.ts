export default class Encounter {
    private fighters: Array<string> = [];

    countFighters(): number {
        return this.fighters.length;
    }

    addFighter(myFighter: string) {
        this.fighters.push(myFighter);
    }
}
