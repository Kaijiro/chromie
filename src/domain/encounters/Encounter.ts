export default class Encounter {
    public readonly fighters: Array<string> = [];
    public readonly unknownLines: Array<string> = [];

    public constructor(public readonly name: string) {
    }

    addFighter(fighter: string) {
        this.fighters.push(fighter);
    }

    addUnknownEvent(eventLine: string) {
        this.unknownLines.push(eventLine);
    }
}
