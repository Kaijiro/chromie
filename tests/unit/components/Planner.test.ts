import {shallowMount} from "@vue/test-utils";
import Planner from "../../../src/components/Planner.vue";
import Encounter from "../../../src/domain/encounters/Encounter";

describe("Planner component", () => {

    const encounter = new Encounter("Boss nÂ°1");
    encounter.addFighter("FighterXYZ");

    xit("should display the list of players of the encounters", () => {
        const mockStore = {
            getters: {
                selectedEncounter: encounter
            }
        };

        const component = shallowMount(Planner, {
            global: {
                mocks: {
                    $store: mockStore
                }
            }
        });

        expect(component["__app"]._container).toMatchSnapshot();
    });

    it("should display the list of enemy of the encounters", () => {

    });

    it("should display the list of technics and spells used by a player when his name is clicked", () => {

    });

    it("should display the list of technics and spells used by an enemy when his name is clicked", () => {

    });
});
