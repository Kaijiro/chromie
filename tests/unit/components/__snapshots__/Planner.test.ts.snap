// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Planner component should display the list of players of the encounters 1`] = `
<div
  data-v-app=""
>
  
   Here we go, let's go on planning.
  <br />
   Planning for Boss n°1
  <br />
   {
  "name": "Boss n°1",
  "fighters": [
    "FighterXYZ"
  ],
  "unknownLines": []
} 
  <table>
    <tr>
      <td>
        <ul>
          
          <li>
            FighterXYZ
          </li>
          
        </ul>
      </td>
    </tr>
  </table>
  
</div>
`;
